### PR TITLE
Restructure for DevEx

### PR DESCRIPTION
**Pull request by Davetbutler**
This PR sets out some structural changes in the repo to make it easier for developers to understand/use.

Setting it as a draft PR as I need to double check READMEs. I have manually tested everything. @Jmgr could you pull this branch and make sure the automated tests are passing please :)

### CODE REVIEW

File: examples_and_tutorials/millionaires_problem_example/02_store_secret_party_n.py [hunk 1]
```diff
@@ -65,15 +65,15 @@ async def main(args = None):
         secret_bindings.add_input_party(party_name, party_id_n)
         print(f"\n🔗 {party_name} sets bindings so that the secret can be input to a specific program (program_id: {args.program_id}) by a specific party (party_id: {party_id_n})")
 
-        # Create permissions object with default permissions for the current user
+        # Create core_concept_permissions object with default core_concept_permissions for the current user
         permissions = nillion.Permissions.default_for_user(user_id_n)
 
-        # Give compute permissions to Alice so she can use the secret in the specific millionionaires program by program id
+        # Give compute core_concept_permissions to Alice so she can use the secret in the specific millionionaires program by program id
```
Comment (oceans404) [inline examples_and_tutorials/millionaires_problem_example/02_store_secret_party_n.py:71]: looks like some comments and descriptions got included in the find and replace of permissions

File: examples_and_tutorials/core_concept_multi_party_compute/02_store_secret_party_n.py [hunk 2]
```diff
@@ -20,7 +20,7 @@
 # N other parties store a secret
 async def main(args = None):
     parser = argparse.ArgumentParser(
-        description="Create a secret on the Nillion network with set read/retrieve permissions"
+        description="Create a secret on the Nillion network with set read/retrieve core_concept_permissions"
```
Comment (oceans404) [inline examples_and_tutorials/core_concept_multi_party_compute/02_store_secret_party_n.py:23]: 👆 permissions find and replace issue

**Review by oceans404 — CHANGES_REQUESTED** [review]
I like the restructure! Some of the find and replace of "permissions" => "core_concept_permissions" should stay the "permissions" since it's part of a comment, not a code path

I flagged these lines/files with "👆 permissions find and replace issue "

File: examples_and_tutorials/core_concept_multi_party_compute/02_store_secret_party_n.py [hunk 3]
```diff
@@ -64,10 +64,10 @@ async def main(args = None):
         secret_bindings = nillion.ProgramBindings(program_id)
         secret_bindings.add_input_party(party_name, party_id_n)
 
-        # Create permissions object
+        # Create core_concept_permissions object
         permissions = nillion.Permissions.default_for_user(user_id_n)
 
-        # Give compute permissions to the first party
+        # Give compute core_concept_permissions to the first party
```
Comment (oceans404) [inline examples_and_tutorials/core_concept_multi_party_compute/02_store_secret_party_n.py:70]: 👆 permissions find and replace issue

File: examples_and_tutorials/core_concept_multi_party_compute/03_multi_party_compute.py [hunk 4]
```diff
@@ -17,15 +17,15 @@
...
+sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), '../..')))
 from helpers.nillion_client_helper import create_nillion_client
 from helpers.nillion_keypath_helper import getUserKeyFromFile, getNodeKeyFromFile
 
 load_dotenv()
 
 async def main(args = None):
     parser = argparse.ArgumentParser(
-        description="Create a secret on the Nillion network with set read/retrieve permissions"
+        description="Create a secret on the Nillion network with set read/retrieve core_concept_permissions"
```
Comment (oceans404) [inline examples_and_tutorials/core_concept_multi_party_compute/03_multi_party_compute.py:28]: 👆 permissions find and replace issue

File: examples_and_tutorials/core_concept_permissions/01_fetch_reader_userid.py [hunk 5]
```diff
@@ -21,7 +21,7 @@ async def main():
     reader_user_id = reader.user_id()
 
     print("ℹ️ Fetched the reader's USER ID:", reader_user_id)
-    print("\n\nRun the following command to store a secret and give read/retrieve permissions to the READER USER ID")
+    print("\n\nRun the following command to store a secret and give read/retrieve core_concept_permissions to the READER USER ID")
```
Comment (oceans404) [inline examples_and_tutorials/core_concept_permissions/01_fetch_reader_userid.py:24]: 👆 permissions find and replace issue

File: examples_and_tutorials/core_concept_permissions/02_store_permissioned_secret.py [hunk 6]
```diff
@@ -16,7 +16,7 @@
 
 async def main(args = None):
     parser = argparse.ArgumentParser(
-        description="Create a secret on the Nillion network with set read/retrieve permissions"
+        description="Create a secret on the Nillion network with set read/retrieve core_concept_permissions"
```
Comment (oceans404) [inline examples_and_tutorials/core_concept_permissions/02_store_permissioned_secret.py:19]: 👆 permissions find and replace issue

File: examples_and_tutorials/core_concept_permissions/02_store_permissioned_secret.py [hunk 7]
```diff
@@ -35,7 +35,7 @@ async def main(args = None):
     writer_user_id = writer.user_id()
     print(writer_user_id, args.retriever_user_id)
 
-    # Writer gives themself default permissions
+    # Writer gives themself default core_concept_permissions
```
Comment (oceans404) [inline examples_and_tutorials/core_concept_permissions/02_store_permissioned_secret.py:38]: 👆 permissions find and replace issue

File: examples_and_tutorials/core_concept_permissions/02_store_permissioned_secret.py [hunk 8]
```diff
@@ -46,7 +46,7 @@ async def main(args = None):
         else "not allowed"
     )
     if result == "not allowed":
-        raise Exception("failed to set permissions")
+        raise Exception("failed to set core_concept_permissions")
```
Comment (oceans404) [inline examples_and_tutorials/core_concept_permissions/02_store_permissioned_secret.py:49]: 👆 permissions find and replace issue

File: examples_and_tutorials/core_concept_permissions/03_retrieve_secret.py [hunk 9]
```diff
@@ -6,15 +6,15 @@
...
+sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), '../..')))
 from helpers.nillion_client_helper import create_nillion_client
 from helpers.nillion_keypath_helper import getUserKeyFromFile, getNodeKeyFromFile
 
 load_dotenv()
 
 async def main(args = None):
     parser = argparse.ArgumentParser(
-        description="Use read permissions to retrieve a secret owned by another user on the Nillion network"
+        description="Use read core_concept_permissions to retrieve a secret owned by another user on the Nillion network"
```
Comment (oceans404) [inline examples_and_tutorials/core_concept_permissions/03_retrieve_secret.py:17]: 👆 permissions find and replace issue

File: examples_and_tutorials/core_concept_permissions/03_retrieve_secret.py [hunk 10]
```diff
@@ -39,7 +39,7 @@ async def main(args = None):
     result = await reader.retrieve_secret(cluster_id, args.store_id, secret_name)
 
     print(f"🦄 Retrieved {secret_name} secret, value = {result[1].value}", file=sys.stderr)
-    print("\n\nRun the following command to revoke the reader's retrieve permissions to the secret")
+    print("\n\nRun the following command to revoke the reader's retrieve core_concept_permissions to the secret")
```
Comment (oceans404) [inline examples_and_tutorials/core_concept_permissions/03_retrieve_secret.py:42]: 👆 permissions find and replace issue

File: examples_and_tutorials/core_concept_permissions/04_revoke_read_permissions.py [hunk 11]
```diff
@@ -7,15 +7,15 @@
...
+sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), '../..')))
 from helpers.nillion_client_helper import create_nillion_client
 from helpers.nillion_keypath_helper import getUserKeyFromFile, getNodeKeyFromFile
 
 load_dotenv()
 
 async def main(args = None):
     parser = argparse.ArgumentParser(
-        description="Revoke user read/retrieve permissions from a secret on the Nillion network"
+        description="Revoke user read/retrieve core_concept_permissions from a secret on the Nillion network"
```
Comment (oceans404) [inline examples_and_tutorials/core_concept_permissions/04_revoke_read_permissions.py:18]: 👆 permissions find and replace issue

**Review by oceans404 — COMMENTED** [review]
I like the restructure! Some of the find and replace of "permissions" => "core_concept_permissions" should stay the "permissions" since it's part of a comment, not a code path

I flagged some of these lines/files with "👆 permissions find and replace issue "

I think the best way to figure out how to update these is do a find and replace of " core_concept_permissions " with a space before and after the word and update back to "permissions" because these are instances where "permissions" is used in a log or comment

**Comment by Davetbutler** [comment]
Thanks @oceans404, must be an over eager pycharm setting...certainly did not do this all manually! Will make sure they are changed when the PR is made properly

**Review by oceans404 — APPROVED** [review]
tested with https://nillion-docs-git-examples-restructure-nillion.vercel.app/

**Pull request merged**